server/prisma/schema.prisma

```diff
@@ -71,42 +71,80 @@ model events {
+  id                     Int                      @id @default(autoincrement())
+  name                   String                   @unique
+  event_role_permissions event_role_permissions[]
+  event_users            event_users[]
+}
+
+model event_role_permissions {
+  created_at          DateTime          @default(now())
+  updated_at          DateTime          @updatedAt
+  event_role_id       Int
+  event_permission_id Int
+  event_role          event_roles       @relation(fields: [event_role_id], references: [id], onDelete: Cascade, onUpdate: NoAction)
+  event_permission    event_permissions @relation(fields: [event_permission_id], references: [id], onDelete: Cascade, onUpdate: NoAction)
+
+  @@id([event_role_id, event_permission_id])
+}
+
+model event_permissions {
+  created_at             DateTime                 @default(now())
+  updated_at             DateTime                 @updatedAt
+  id                     Int                      @id @default(autoincrement())
+  name                   String                   @unique
+  event_role_permissions event_role_permissions[]
+}
+
+model event_users {
+  created_at      DateTime  @default(now())
+  updated_at      DateTime  @updatedAt
+  user_id         Int
+  event_id        Int
+  event_role_id   Int
+  rsvp_id         Int
+  subscribed      Boolean
+  title           String[]
+  user            users       @relation(fields: [user_id], references: [id], onDelete: Cascade, onUpdate: NoAction)
+  event           events      @relation(fields: [event_id], references: [id], onDelete: Cascade, onUpdate: NoAction)
+  event_role      event_roles @relation(fields: [event_role_id], references: [id], onDelete: NoAction, onUpdate: NoAction)
+  rsvp            rsvp    @relation(fields: [rsvp_id], references: [id], onDelete: NoAction, onUpdate: NoAction)
 
   @@id([user_id, event_id])
 }
 
 model event_reminders {
-  created_at    DateTime  @default(now())
-  updated_at    DateTime  @updatedAt
+  created_at    DateTime    @default(now())
+  updated_at    DateTime    @updatedAt
   user_id       Int
   event_id      Int
+  rsvp_id       Int
   remind_at     DateTime
-  notifying     Boolean   @default(false)
-  event         events    @relation(fields: [event_id], references: [id], onDelete: Cascade, onUpdate: NoAction)
-  user          users     @relation(fields: [user_id], references: [id], onDelete: Cascade, onUpdate: NoAction)
-  rsvp          rsvps     @relation(fields: [user_id, event_id], references: [user_id, event_id], onDelete: Cascade, onUpdate: NoAction)
+  notifying     Boolean     @default(false)
+  event         events      @relation(fields: [event_id], references: [id], onDelete: Cascade, onUpdate: NoAction)
+  user          users       @relation(fields: [user_id], references: [id], onDelete: Cascade, onUpdate: NoAction)
+  rsvp          rsvp        @relation(fields: [rsvp_id], references: [id], onDelete: Cascade, onUpdate: NoAction)
```

Comment:
Okay, so reminders can only exist if there's an `event_user` because if someone has no relationship to an event, they definitely don't want a reminder.  To me, it makes most sense to go with your idea of replacing all three with a single `event_user` reference.

I've been messing with this locally, do you mind if I push a commit?

Reply:
Not at all, go ahead. :+1: 

But make sure you have the latest changes, I was correcting merge conflicts earlier today.

Reply:
Thanks for the warning, I'll double-check!